### PR TITLE
[console] Organization audit

### PR DESCRIPTION
**Pull request by ThibaudAV**
**Issue**
https://github.com/gravitee-io/issues/issues/7536

**Description**
After this backend stuff : 
https://github.com/gravitee-io/gravitee-api-management/pull/1659

Add new screen for organization audit

**Additional context**
image

🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/7521-audit-org/index.html)
_Notes_: The deployed app is linked to the management API of APIM master. (Same login and password as APIM master)

### CODE REVIEW

File: gravitee-apim-console-webui/src/services-ngx/audit.service.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import { MetadataPage } from '../entities/MetadataPage';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuditService {
+  constructor(private readonly http: HttpClient, @Inject('Constants') private readonly constants: Constants) {}
+
+  listByOrganization(): Observable<MetadataPage<Audit>> {
+    return this.http.get<MetadataPage<Audit>>(`${this.constants.org.baseURL}/audit`);
```
Comment (gaetanmaisse) [inline gravitee-apim-console-webui/src/services-ngx/audit.service.ts]: I think it should be plural here `/audits` and so here too https://github.com/gravitee-io/gravitee-api-management/pull/1659/files#diff-ae8e4ed2b9141528c8dbec5a2e3e70893e64f60804b68c387c77d75c38540527R224 no?

Comment (gaetanmaisse) [reply gravitee-apim-console-webui/src/services-ngx/audit.service.ts]: Ok, it's consistant with the other url (ApiAudit) but it's weird 🤔

Comment (ThibaudAV) [reply gravitee-apim-console-webui/src/services-ngx/audit.service.ts]: not strange to have plural only for org audits ?

Comment (ThibaudAV) [reply gravitee-apim-console-webui/src/services-ngx/audit.service.ts]: Yes 🤷‍♂️

File: gravitee-apim-console-webui/src/organization/configuration/audit/org-settings-audit.component.spec.ts [hunk 2]
```diff
@@ -80,13 +80,29 @@ describe('OrgSettingsAuditComponent', () => {
     expectAuditListRequest({ event: 'ROLE_UPDATED' });
   });
 
+  it('should display audit logs with type filter', async () => {
+    expectAuditListRequest();
+    expectAuditEventsNameRequest();
+
+    const table = await loader.getHarness(MatTableHarness.with({ selector: '#auditTable' }));
+    const rows = await table.getRows();
+    const rowCells = await await parallel(() => rows.map((row) => row.getCellTextByColumnName()));
```
Comment (gaetanmaisse) [inline gravitee-apim-console-webui/src/organization/configuration/audit/org-settings-audit.component.spec.ts]: `await await` 🦉

File: gravitee-apim-console-webui/src/organization/configuration/audit/org-settings-audit.component.ts [hunk 3]
```diff
@@ -58,24 +58,30 @@ export class OrgSettingsAuditComponent implements OnInit, OnDestroy {
...
       .pipe(
         takeUntil(this.unsubscribe$),
         throttleTime(100),
         distinctUntilChanged(),
-        switchMap(({ pagination, event }) => this.auditService.listByOrganization({ event }, pagination.index, pagination.size)),
+        switchMap(({ pagination, event, referenceType }) =>
+          this.auditService.listByOrganization({ event, referenceType }, pagination.index, pagination.size),
+        ),
+        catchError(() => {
+          this.snackBarService.error('Unable to try the request, please try again');
```
Comment (gaetanmaisse) [inline gravitee-apim-console-webui/src/organization/configuration/audit/org-settings-audit.component.ts]: `Unable to run the request, please try again`

File: gravitee-apim-console-webui/src/services-ngx/audit.service.spec.ts [hunk 4]
```diff
@@ -35,19 +35,49 @@ describe('AuditService', () => {
...
+
+      auditService.listByOrganization({ event: 'HELLO' }).subscribe((response) => {
+        expect(response).toEqual(fakeAuditPage);
+        done();
+      });
+
+      const req = httpTestingController.expectOne(`${CONSTANTS_TESTING.org.baseURL}/audit?page=1&size=10&event=HELLO`);
+      expect(req.request.method).toEqual('GET');
+
+      req.flush(fakeAuditPage);
```
Comment (gaetanmaisse) [inline gravitee-apim-console-webui/src/services-ngx/audit.service.spec.ts]: In later commits we are using another syntax, I think we should be consistant: 
```ts
      httpTestingController.expectOne({
        method: 'GET',
        url: `${CONSTANTS_TESTING.org.baseURL}/audit?page=1&size=10&event=HELLO`
      }).flush(fakeAuditPage);
```

File: gravitee-apim-console-webui/src/services-ngx/audit.service.spec.ts [hunk 5]
```diff
@@ -35,19 +35,49 @@ describe('AuditService', () => {
...
+
+      auditService.getAllEventsNameByOrganization().subscribe((response) => {
+        expect(response).toEqual(fakeEvents);
+        done();
+      });
+
+      const req = httpTestingController.expectOne(`${CONSTANTS_TESTING.org.baseURL}/audit/events`);
+      expect(req.request.method).toEqual('GET');
+
+      req.flush(fakeEvents);
```
Comment (gaetanmaisse) [inline gravitee-apim-console-webui/src/services-ngx/audit.service.spec.ts]: Same here

File: gravitee-apim-console-webui/src/organization/configuration/audit/org-settings-audit.component.ts [hunk 6]
```diff
@@ -54,29 +63,81 @@ export class OrgSettingsAuditComponent implements OnInit, OnDestroy {
...
+            ...res,
+            [env.name]: this.apiService.getAll({
+              environmentId: env.id,
+            }),
+          }),
+          {} as Record<string, Observable<Api[]>>,
+        ),
+      ),
+    ),
+  );
```
Comment (gaetanmaisse) [inline gravitee-apim-console-webui/src/organization/configuration/audit/org-settings-audit.component.ts:82]: 🤯

File: gravitee-apim-console-webui/src/organization/configuration/audit/org-settings-audit.component.spec.ts [hunk 7]
```diff
@@ -125,19 +126,48 @@ describe('OrgSettingsAuditComponent', () => {
...
+    expectAuditListRequest();
+    expectAuditEventsNameRequest();
+
+    // 1. Expect initial page
+    const table = await loader.getHarness(MatTableHarness.with({ selector: '#auditTable' }));
+    const rows = await table.getRows();
+    const rowCells = await await parallel(() => rows.map((row) => row.getCellTextByColumnName()));
+    expect(rowCells.length).toEqual(20);
+
+    // 3. Expect date range works and trigger new AuditListRequest
```
Comment (gaetanmaisse) [inline gravitee-apim-console-webui/src/organization/configuration/audit/org-settings-audit.component.spec.ts]: No `2.`

**Review by gaetanmaisse — APPROVED** [review]
I think validation of both a UI/UX and PO would be great as we have 1 new screen and 1 migration

**Comment by mouligno** [comment]
> I think validation of both a UI/UX and PO would be great as we have 1 new screen and 1 migration

I've checked this screen with @ThibaudAV while he was building it !

File: gravitee-apim-console-webui/src/organization/configuration/audit/org-settings-audit.component.ts (hunk 6 above)
Comment (ThibaudAV) [reply gravitee-apim-console-webui/src/organization/configuration/audit/org-settings-audit.component.ts:82]: i've add some comment

**Review by gaetanmaisse — APPROVED** [review]
🚀

**Pull request merged**